lib/auth/github.go

```diff
@@ -334,6 +334,28 @@ type createUserParams struct {
 	sessionTTL time.Duration
 }
 
+func (params createUserParams) toMap() map[string]interface{} {
```

Comment:
Not sure why the params needs to be converted to map to be forwarded to `TraceSAMLDiagnosticInfo` as `extraInfo` param where `extraInfo` has the `interface{}` type.

Reply:
This is a workaround really as `createUserParams` has no exported fields and I don't want to change that type as it is used very widely.

Reply:
The function `toMap()` is no more, instead `createUserParams` has custom JSON marshaller now.